src/components/AttachmentModal.tsx

```diff
@@ -279,19 +308,20 @@ function AttachmentModal(props) {
      * @param {Object} _data
      */
     const validateAndDisplayFileToUpload = useCallback(
-        (_data) => {
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        (_data: Data) => {
             if (!isDirectoryCheck(_data)) {
                 return;
             }
-            let fileObject = _data;
-            if (typeof _data.getAsFile === 'function') {
+            let fileObject: Data | null = _data;
+            if ('getAsFile' in _data && typeof _data.getAsFile === 'function') {
                 fileObject = _data.getAsFile();
             }
             if (!fileObject) {
                 return;
             }
 
-            if (!isValidFile(fileObject)) {
+            if (fileObject instanceof File && !isValidFile(fileObject)) {
```

Comment:
Are you sure this condition is accurate? I think you can update `isValidFile` props typing instead